launcher/ui/pages/instance/DataPackPage.cpp

```diff
@@ -65,9 +67,23 @@ void DataPackPage::downloadDataPacks()
     if (m_instance->typeName() != "Minecraft")
         return;  // this is a null instance or a legacy instance
 
-    ResourceDownload::DataPackDownloadDialog mdownload(this, std::static_pointer_cast<DataPackFolderModel>(m_model), m_instance);
-    if (mdownload.exec()) {
-        auto tasks = new ConcurrentTask("Download Data Pack", APPLICATION->settings()->get("NumberOfConcurrentDownloads").toInt());
+    auto profile = static_cast<MinecraftInstance*>(m_instance)->getPackProfile();
```

Comment:
do datapacks *need* a mod loader>

Reply:
Trial made the same criticism. and I fiddled about and could not really find another good solution.

I said I would reopen if he came up with a better solution but that never happened – I think my solution ultimately ended up requiring many code changes and not even making that much sense.

Reply:
I'm confused. the reason to use datapacks is often because they don't need a mod loader. what about our internal stuff requires a mod loader to allow management of data packs?

if such a requirement exists I would consider that a blocker for merging.

Reply:
a bit of context from previous discussions https://github.com/PrismLauncher/PrismLauncher/pull/1905#discussion_r2013580746
modrinth considers somehow datapacks as a loader or something

Reply:
> imo Adding a datapack should be no different than adding a resource pack ux wise

Indeed - it is the same in terms of UX
It is just in the loaders as modrinth considers it a loader and it makes it much easier to pass it through the code

Reply:
If you don't want to read my descent into insanity I can summarise it as this - though it has been a while since i thought about this:
- Modrinth has a datapack project type and a datapack loader.
- If you only want datapacks you need to filter the project search by both the datapack project type and data pack loader
- You also need to filter by the data pack loader on the versions - so the easiest option is to just add this to the ModLoader enum so it automatically applies to both.
- I tried more elaborate options like passing the project type into `VersionSearchArgs` too, or a boolean - but both of those values would only do something in very specific circumstances so it didn't feel worth it - and it also required changing a large amount of duplicated code to pass this data around

Reply:
@Ryex hello